src/powershell/Public/ConvertTo-FinOpsSchema.ps1

```diff
@@ -0,0 +1,214 @@
+# If the value is not the same as the key, the column will be renamed and the value will be used as the new column name.
+# If the key is not present in the input data, the column will not be included in the output data.
+$columnMapping = @{
+    billingAccountId = 'BillingAccountId'
+    billingAccountName = 'BillingAccountName'
+    ftk_AccountType = 'ftk_AccountType'
+    BillingProfileId = 'BillingProfileId'
+    SubscriptionId = 'SubscriptionId'
+}
+
+<# Helper Functions #>
+function Add-AccountType {
+    [CmdletBinding()]
+    param 
+    (
+        [PSCustomObject]
+        $Object
+    )
+
+    $ftk_AccountType = switch ($true) {
+        { $Object.BillingAccountId -eq $Object.BillingProfileId } { "EA" }
+        { $null -eq $Object.BillingProfileId -or 
+          $Object.BillingProfileId -eq $Object.SubscriptionId } { "Other" }
+        default { "MCA" }
+    }
+    # Add or Update the ftk_AccountType property to the object.
+    if ($Object.PSObject.Properties["ftk_AccountType"]) {
+        $Object.PSObject.Properties["ftk_AccountType"].Value = $ftk_AccountType
+    } else {
+        $Object | Add-Member -MemberType NoteProperty -Name "ftk_AccountType" -Value $ftk_AccountType -Force
+    }
+    
+    return $Object
+}
+
+
+<# Main Function #>
+function ConvertTo-FinOpsSchema {
+    [CmdletBinding()]
+    param
+    (
+        [Parameter()]
+        [string]
+        $ActualCost,
+
+        [Parameter()]
+        [string]
+        $outputFilePath,
+
+        [Parameter()]
+        [hashtable]
+        $columnMapping,
+
+        [Parameter()]
+        [bool]
+        $ExportAllColumns,
+
+        [Parameter()]
+        [switch]
+        $Force
+    )
+    
+    if (-not (Test-Path -Path $ActualCost)) {
+        Write-Error "Input file $ActualCost does not exist."
+        return
+    }
+    
+    $outputDirectory = [System.IO.Path]::GetDirectoryName($outputFilePath)
+    if (-not (Test-Path -Path $outputDirectory -PathType Container)) {
+        New-Item -Path $outputDirectory -ItemType Directory
+    }
+
+    # Import the CSV data
+    $csvData = Import-Csv -Path $ActualCost
+    
+    # Iterate over each row and apply transformations
+    $transformedData = $csvData | ForEach-Object {
+        $object = $_
+        $object = Add-AccountType -Object $object
+
+        # Create a new object with the mapped column names
+        $newObject = @{}
+        foreach ($column in $object.PSObject.Properties.Name) {
+            if ($ExportAllColumns -or $column -in $columnMapping.Keys) {
+                $newColumnName = if ($columnMapping.ContainsKey($column)) { $columnMapping[$column] } else { $column }
+                $newObject[$newColumnName] = $object.$column
+            }
+        }
+
+        # Output the new object
+        [PSCustomObject]$newObject
+    }
+    
+    # Export the transformed data to the output CSV file
+    $transformedData | Export-Csv -Path $outputFilePath -NoTypeInformation -Encoding UTF8
```

Comment:
My preference would be to not include this in this function. If you want to do that, we should have a separate `Export-FinOpsSchema` function that calls `ConverTo-FinOpsSchema`.

Reply:
removed the whole code with Export-Csv after the pipesymbol